### PR TITLE
feat( GoogleService (38 > main)) create google service

### LINKED ISSUE #38: Create Google service
- [ ] activate the service we want on google cloud (ask @FoxaxeWasTaken) for example

Image

- [ ] add the service on the DB with :
       - userId
       -  token from the service provider
       -  enum of the service

### PR DESCRIPTION
**Pull request by Pablodeibar**
Closes #38 
create a service like that on DB with the access token for Google API
image

### CODE REVIEW

File: server/src/google-auth/google-auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  @Get()
+  @UseGuards(GoogleOAuthGuard)
+  async googleAuth(@Request() req) {} // eslint-disable-line
+
+  @Get('google-redirect')
+  @UseGuards(GoogleOAuthGuard)
+  async googleAuthRedirect(@Request() req) {
+    const token = await this.appService.googleLogin(req);
+    console.log(token);
```
Comment (Benjicatch) [inline server/src/google-auth/google-auth.controller.ts:23]: remove console.log

File: server/src/google-auth/google-auth.controller.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    private readonly appService: GoogleAuthService,
+    private prisma: PrismaService,
+  ) {}
+
+  @Get()
+  @UseGuards(GoogleOAuthGuard)
+  async googleAuth(@Request() req) {} // eslint-disable-line
+
+  @Get('google-redirect')
+  @UseGuards(GoogleOAuthGuard)
```
Comment (Benjicatch) [inline server/src/google-auth/google-auth.controller.ts:20]:
```suggestion
  @UseGuards(GoogleOAuthGuard)
  @ApiOperation({
      summary: 'what this route do'
  })
```

**Review by Benjicatch — CHANGES_REQUESTED** [review]

**Review by Benjicatch — APPROVED** [review]
ok

**Review by AlexGuillard — APPROVED** [review]
ok

**Pull request merged**